### DISCUSSION TITLE
Request for Help: Inconsistencies Between EBSD Maps and ODFs for the Same .ctf File

### DISCUSSION THREAD

**Discussion by leonelmtex** [mtex-toolbox/mtex#2205] [Ask Anything]
Hello everyone,

I'm currently working on a revision of a paper and have encountered an issue with inconsistencies between EBSD maps generated from the same .ctf file. Specifically, I'm seeing different results in both the EBSD maps and the Orientation Distribution Functions (ODFs) when using the same .ctf file but processed in two different ways.

To illustrate this, I've attached two images:

A comparison between the EBSD maps: one generated by me using [name of the software/tool you used] and another automatically generated by the SEM software.
A comparison of the ODFs generated from each of these EBSD maps.
I would like to understand why I'm seeing these differences. Could there be an issue with the processing or settings? What might be the causes of these inconsistencies?

I'm attaching the original .ctf file in case anyone is willing to check it and help identify what might be causing these discrepancies. Any guidance or suggestions on how to investigate and resolve this issue would be greatly appreciated, especially as I'm in the process of revising a paper.

Thank you for your help!

Attachments:
![1](https://github.com/user-attachments/assets/9976ade6-3739-4b41-8418-ae6cac98ba9a)
![2](https://github.com/user-attachments/assets/ca6ee269-52e0-4a01-bad4-0b6893d756ff)

Original .ctf file. https://drive.google.com/file/d/1IQ7io-bjoPhW-D8ACCGO25gNNqmTFUiR/view?usp=sharing

```matlab
% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [2.9 2.9 2.9], 'mineral', 'Iron bcc (old)', 'color', [0.53 0.81 0.98]),...
  crystalSymmetry('m-3m', [3.7 3.7 3.7], 'mineral', 'Iron fcc', 'color', [0.56 0.74 0.56])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','intoPlane');

%% Specify File Names

% path to files
pname = 'C:\Users...';

% which files to be imported
fname = [pname '\23087-C 300X st3 bin8x8 fr1 sum.ctf'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ctf');


%%
plot(ebsd, ebsd.orientations)
%%
% extract the orientations
ori = ebsd.orientations;

% compute the ODF
odf = calcDensity(ori)
plot(odf)
```

**Comment by zmichels** [comment]
What import settings are you using? And do you apply any rotations at import (like in an import script/code-snippet)? -- If not, it might help to align the orientation reference frame, and I can help.

Without knowing a bit more, I can't say for certain... but I think the issue is a common one due to acquisition-system/software settings... in which the spatial coordinates for the position of points may not utilize the same coordinate system as the camera+solution-engine (i.e., maybe the spatial coordinates are relative to the stage position, whereas the orientation coordinate system is based on the camera's viewpoint of the stage). 

Your comparison plots [comparing between MTEX current configuration and acquisition system] are very diagnostic already, and exactly the type of approach I use whenever I get new data or collect on a new system... although I often use pole figures instead/also. I am confident that if you can share more about your plotting convention preferences and/or rotations you have set in mtex_settings.m or at import of the data, we'll be able to get them to match.

Zach

**Reply by leonelmtex** [reply]
Thank you for your help!
I suspected that this might be the issue, and this complicates the situation further. These data were obtained by a collaborator at a research center within a company, but he no longer has access to the software as he has changed jobs, and only the measured files (.ctf) remain.

I am investigating texture properties after a specific heat treatment. Specifically, I am looking into whether the material has transitioned to Goss, Alpha, Gamma, or Etha textures after the treatment. For this, I am analyzing ODF maps (Bunge diagram).

At the moment of data import, I am using the settings shown in the figure.
![image](https://github.com/user-attachments/assets/961da08a-6ad5-4b03-8e32-bd4cbbdd3b48)

**Comment by zmichels** [comment]
Also... I tried the link for the CTF, and I was not allowed access. Do you mind changing the permissions?

**Comment by zmichels** [comment]
I was able to get the spatial plot to match yours coupled with a flipped (compared to the posted one) ODF section plot. I am not certain it matches the ODF section from acquisition system because there are a number of differences... but the key is that I used a combination of plotting preference and rotations at import to change the alignment. Note/FWIW: typically, unless the system is altered or reconfigured, once I get a settled import+plotting solution for data from a given lab, I can use that on all the files from that lab. The following is the import code I tried and the resultant plots:

```matlab
%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = {... 
  'notIndexed',...
  crystalSymmetry('m-3m', [2.9 2.9 2.9], 'mineral', 'Iron bcc', 'color', [1 0.6 0]),...
  crystalSymmetry('m-3m', [3.7 3.7 3.7], 'mineral', 'Iron fcc', 'color', [0.6 0.8 1])};

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = 'CHANGE_TO_YOUR_PATH';

% which files to be imported
fname = [pname '/LM_CTF_example.ctf'];

%% Import the Data

% create an EBSD variable containing the data
ebsd = EBSD.load(fname,CS,'interface','ctf',...
  'convertEuler2SpatialReferenceFrame');


%% some rotations to try
ebsd = rotate(ebsd,rotation('axis',yvector,'angle',180*degree));
ebsd = rotate(ebsd,rotation('axis',zvector,'angle',180*degree));

```

![LM_CTF_example_with_rots_XYZ_ebsd](https://github.com/user-attachments/assets/c25fda47-5c17-4e53-96f7-931c34fa3539)
![LM_CTF_example_with_rots_XYZ_ODF](https://github.com/user-attachments/assets/868dd244-cf7f-430c-8929-5fd4e1abb8b7)
![LM_CTF_example_with_rots_XYZ_PF](https://github.com/user-attachments/assets/91b68a32-35eb-418c-93d0-6c437010cd5b)

**Reply by leonelmtex** [reply]
Thank you for the excellent help! Indeed, the data doesn't match exactly. However, I used another software, ATEX, and it returned a result much closer to the file received from the lab's software, as long as I set the sample symmetry to orthorhombic.

I tested this with ATEX, but I had to specify that it was orthorhombic. After discussing with my collaborator, he confirmed that the software he used at the time also had the orthorhombic setting, and in the software, only the options for monoclinic and triclinic were available. Luckily, he had an image on his WhatsApp, attached, where more information can be seen.

I have also attached the ODF maps obtained with ATEX. My problem is resolved to some extent, but I want to advance my knowledge of MTEX.

Do you have any observations or tips? Thank you again!

![23087-C _Euler_All](https://github.com/user-attachments/assets/8b2c29e0-b42c-4ca0-8011-0d2e83ca4947)

![image](https://github.com/user-attachments/assets/0cf37da1-8f95-4ea8-b6f3-966bef271c98)
